### ISSUE TITLE
scf.pl.trends error on jupyter notebook

### ISSUE THREAD

**Issue by hw1804** [LouisFaure/scFates#21]
<img width="1118" alt="Screenshot 2023-08-18 at 17 01 27" src="https://github.com/LouisFaure/scFates/assets/105513480/ddecbf22-4a1f-4ac4-b708-b962dcd02822">

Hi, I was trying to plot a series of graph using scf.pl.trends and I have done other procedures in the basic curved trajectory analysis tutorial "fitting&clustering significant features" part. I am using the tutorial code and it gave the error in the screenshot: "arrays used as indices must be of integer (or boolean) type". Is there any work-around of this issue?

**Comment by hw1804** [comment]
I was running my own dataset using the tutorial, that's probably why it has got 9 clusters. I ran the test code, it gave [1] object instead of [0] int. One of my friend thinks it may be because of the python version I am using. I am using python 3.11.4. I am not sure why my python or notepad interact differently.  
<img width="1153" alt="Screenshot 2023-08-22 at 22 11 27" src="https://github.com/LouisFaure/scFates/assets/105513480/e86a7c92-c763-4aae-9de5-be23738c301b">

**Comment by LouisFaure** [comment]
I just tried with python 3.11 and I am not getting this issue (and `order` is numpy array with int64), this could be dataset related. Do you see the same error by reproducing exactly the [example notebook](https://github.com/LouisFaure/scFates_notebooks/blob/main/Basic_Curved_trajectory_analysis.ipynb)?

**Comment by hw1804** [comment]
I ran the tutorial with your dataset and was able to get the scf.pl.trends to work. I ran the same code again using my dataset and it still gave the same error. My dataset was originally a seurat object, and I converted it to h5ad file for python packages to read. Any pointers on how can I fix my dataset? I could plot a lot of graphs fine, just not the scf.pl.trends, which I really want to show for publication.
<img width="1141" alt="Screenshot 2023-08-23 at 12 14 43" src="https://github.com/LouisFaure/scFates/assets/105513480/0f3d75df-46f5-414b-93bd-be374e5f562f">
<img width="1134" alt="Screenshot 2023-08-23 at 12 15 00" src="https://github.com/LouisFaure/scFates/assets/105513480/01b97051-15bd-4bf3-bfcf-bc8bdefb710f">

**Comment by hw1804** [comment]
update: I can plot scf.pl.single_trend just fine. But scf.pl.trends not working. I have converted my seurat file to loom and still the same error. Here is what my dataset look like:
<img width="1104" alt="Screenshot 2023-08-23 at 15 12 23" src="https://github.com/LouisFaure/scFates/assets/105513480/4742e439-1694-42db-a4d7-6ce2c139acd8">
<img width="582" alt="Screenshot 2023-08-23 at 15 10 57" src="https://github.com/LouisFaure/scFates/assets/105513480/f76a07a8-3ad0-4ef7-a556-a79b0db0c764">

**Comment by LouisFaure** [comment]
Could you try this code?
```python
import igraph
import numpy as np
import pandas as pd
graph=adata.uns["graph"]
dct = graph["milestones"]
keys = np.array(list(dct.keys()))
vals = np.array(list(dct.values()))
edges = graph["pp_seg"][["from", "to"]].astype(str).apply(tuple, axis=1).values
img = igraph.Graph(directed=True)
img.add_vertices(vals.astype(str))
img.add_edges(edges)

allpaths = img.get_all_shortest_paths(
    str(graph["root"]), to=graph["tips"].astype(str)
)

allpaths = np.array(allpaths, dtype=object)[
    np.argsort(np.array([len(p) for p in allpaths]))
]

order = allpaths[0]
for i in range(1, len(allpaths)):
    order = order + np.array(allpaths[i])[~np.isin(allpaths[i], order)].tolist()

order = np.array(order)[1:]
print(pd.Series(graph["milestones"].keys(), index=graph["milestones"].values())[
    [np.array(img.vs["name"])[o].astype(int) for o in order]
])
```
If it succeed I will probably be able to change the part of the function that was causing the error, avoid the check for int or booleans

**Comment by hw1804** [comment]
<img width="1124" alt="Screenshot 2023-08-23 at 22 01 10" src="https://github.com/LouisFaure/scFates/assets/105513480/94f91e76-6488-4d67-b0bc-fd89d4b06b97">

**Comment by LouisFaure** [comment]
I think we are getting there! Could you try now to use `pl.trends`using that version of scFates:
```bash
pip install git+https://github.com/LouisFaure/scFates@v1.0.5dev
```

**Comment by hw1804** [comment]
Thank you so much for your help. The pl.trends works now for my dataset as well with the dev version.

**Comment by LouisFaure** [comment]
Great! I have now included the changes to v1.0.5
